src/api/util/ordinal-satoshi.ts

```diff
@@ -75,7 +81,8 @@ export class OrdinalSatoshi {
   }
 
   public get percentile(): string {
-    return `${(this.ordinal / (SAT_SUPPLY - 1)) * 100.0}%`;
+    const percentile = new BigNumber((this.ordinal / (SAT_SUPPLY - 1)) * 100.0);
+    return `${percentile.toFixed()}%`;
```

Comment:
I'm still sus on the floating point operations performed within the BigNumber constructor, i.e. seems like the division and multiplication should be performed with BigNumber operations. Given the number of decimals I'd be pretty surprised if there were not floating point errors. 

But I don't know how important it is to have this `percentile` value exactly accurate, up to you

Reply:
I think you're right that we could have more precision with `BigNumber` operations but we'd be moving away from the canonical source of percentile which is ordinals.com.

To give you an example, I modified this function to use BigNumber calculations like so:
```typescript
  public get percentile(): string {
    const percentile = new BigNumber(this.ordinal)
      .dividedBy(new BigNumber(SAT_SUPPLY - 1))
      .multipliedBy(100.0);
    return `${percentile.toFixed()}%`;
  }
```

This greatly increased the precision of the calculation, but now we have a "different" number than https://ordinals.com/sat/2067187500000000:
<img width="463" alt="Screenshot 2023-05-12 at 10 12 20" src="https://github.com/hirosystems/ordinals-api/assets/216195/a8e9cf71-b94d-436f-bbbe-7edbbfe89f4f">

In this case, since we're keeping our data as consistent as we can with ordinals.com I think we should leave it as it is.

Reply:
Makes sense to me, thanks!